packages/babel-parser/src/parser/util.js

```diff
@@ -35,18 +34,6 @@ export default class UtilParser extends Tokenizer {
     return this.match(tt.relational) && this.state.value === op;
   }
 
-  isLookaheadRelational(op: "<" | ">"): boolean {
-    const next = this.nextTokenStart();
-    if (this.input.charAt(next) === op) {
-      if (next + 1 === this.input.length) {
```

Comment:
This check is removed in `isLookaheadToken_lt` because if `next + 1` is the `length`, `afterNext` is `NaN`, thus it always satisfies
```
afterNext !== charCodes.lessThan && afterNext !== charCodes.equalsTo
```